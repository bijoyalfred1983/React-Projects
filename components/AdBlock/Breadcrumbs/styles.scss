@import '../../theme/variables';

.homeLink, .homeLink:hover {
  color:  $az-crust-grey !important;
  font-style: normal !important;
  font-weight:  normal !important;
  font-size:  12px !important;
  background: transparent;
  border: none;
  padding: 0;
  font-family: $font-az-app;
}

.activeLink  {
  font-size:  12px !important;
  color:  $crust-white-color !important;
  font-style:  normal !important;
  font-weight:  normal !important;
}

.dividerLink  {
  opacity:  1 !important;
  font-size:  20px !important;
  color:  $az-crust-grey !important;
  height:  12px !important;
  margin-right: 5px !important;
  font-weight: lighter !important;
  background:  transparent !important;
}

.breadcrumbLinkAZ  {
  padding: 0 !important;
  margin:20px !important;
}

.mobileCrumb  {
  display:  none !important;

  @media #{$extra-large-and-down} {
    display: block !important;
  }
}

.desktopCrumb  {
  @media #{$extra-large-and-down}{
    display: none !important;
  }
}

