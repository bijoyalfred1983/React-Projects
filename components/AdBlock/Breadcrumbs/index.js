/* @flow */
import React, { PureComponent } from 'react';
import * as styles from './styles.scss';
import MobileCrumb from './MobileCrumb/mobileCrumbComponent';
import DesktopCrumb from './DesktopCrumb/desktopCrumbComponent';

type Props = {
  data: any
};

class Breadcrumbs extends PureComponent<Props> {
  render() {
    const breadcrumbsList = this.props.data;
    if (breadcrumbsList && breadcrumbsList.length > 0) {
      return (
        <div>
          <div className={styles.desktopCrumb}>
            <DesktopCrumb data={breadcrumbsList} />
          </div>
          <div className={styles.mobileCrumb}>
            <MobileCrumb data={breadcrumbsList} />
          </div>
        </div>
      );
    }
    return null;
  }
}

export default Breadcrumbs;
