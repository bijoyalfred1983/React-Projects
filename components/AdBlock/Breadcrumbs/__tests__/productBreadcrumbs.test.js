import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Breadcrumbs from '../index';

describe('<Breadcrumbs />', () => {
  let PROPS;
  beforeEach(() => {
    PROPS = {
      data: [
        {
          '@class': 'com.autozone.diy.vo.AZBreadcrumbVO',
          displayName: 'Parts',
          url: '/Parts'
        },
        {
          '@class': 'com.autozone.diy.vo.AZBreadcrumbVO',
          displayName: 'Brakes And Traction Control',
          url: '/parts/brakes-and-traction-control'
        },
        {
          '@class': 'com.autozone.diy.vo.AZBreadcrumbVO',
          displayName: 'Disc Brake System',
          url: '/parts/brakes-and-traction-control/disc-brake-system'
        },
        {
          '@class': 'com.autozone.diy.vo.AZBreadcrumbVO',
          displayName: 'Brake Pads',
          url: '/brakes-and-traction-control/brake-pads'
        },
        {
          '@class': 'com.autozone.diy.vo.AZBreadcrumbVO',
          displayName: 'Duralast PlatinumBattery_BT',
          url: ''
        }
      ]
    };
  });

  test('renders on breadcrumbs correctly', () => {
    const wrapper = mount(<Breadcrumbs {...PROPS} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('renders error when breadcrumbs is empty', () => {
    const LOCAL_PROPS = {
      data: []
    };
    const wrapper = mount(<Breadcrumbs {...LOCAL_PROPS} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
