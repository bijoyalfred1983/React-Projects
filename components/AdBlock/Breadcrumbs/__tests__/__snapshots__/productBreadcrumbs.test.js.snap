// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductBreadcrumbs /> renders error when breadcrumbs is empty 1`] = `
<ProductBreadcrumbs
  data={Array []}
/>
`;
